src/message/MoveMessage.js

```diff
@@ -0,0 +1,216 @@
+  const allStreams = useSelector(getStreams);
+  const isAdmin = useSelector(getOwnUser).is_admin;
+  const messageId = props.route.params.message.id;
+  const currentStreamName = streamNameOfNarrow(props.route.params.messageNarrow);
+  const currentStreamId = allStreams.find(s => s.name === currentStreamName)?.stream_id;
+  const [narrow, setNarrow] = useState(streamNarrow(currentStreamName));
+  const [subject, setSubject] = useState(props.route.params.message.subject);
+  const [propagateMode, setPropagateMode] = useState('change_one');
+  const [streamId, setStreamId] = useState(currentStreamId);
+  const [topicFocus, setTopicFocus] = useState(false);
+
+  const styles = {
+    parent: {
+      backgroundColor: cardColor,
+    },
+    layout: {
+      margin: 10,
+    },
+    title: {
+      fontSize: 18,
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    topicInput: {
+      height: 50,
+      backgroundColor,
+      ...inputMarginPadding,
+    },
+    viewTitle: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      height: 50,
+      paddingHorizontal: 10,
+      marginBottom: 20,
+    },
+    textColor: {
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    picker: { backgroundColor, marginBottom: 20 },
+    submitButton: {
+      marginTop: 10,
+      paddingTop: 15,
+      paddingBottom: 15,
+      marginLeft: 30,
+      marginRight: 30,
+      backgroundColor: BRAND_COLOR,
+      borderRadius: 10,
+      borderWidth: 1,
+    },
+  };
+
+  const handleTopicChange = (topic: string) => {
+    setSubject(topic);
+  };
+
+  const handleTopicFocus = () => {
+    setTopicFocus(true);
+  };
+
+  const setTopicInputValue = (topic: string) => {
+    handleTopicChange(topic);
+    setTopicFocus(false);
+  };
+
+  const handleTopicAutocomplete = (topic: string) => {
+    setTopicInputValue(topic);
+  };
+
+  const updateMessage = () => {
+    if (isAdmin) {
+      api
+        .updateMessage(
+          auth,
+          { subject, stream_id: streamId, propagate_mode: propagateMode },
+          messageId,
+        )
+        .catch(error => {
+          showErrorAlert('Failed to move message', error.message);
+          props.navigation.goBack();
+        });
+    } else {
+      api
+        .updateMessage(auth, { subject, propagate_mode: propagateMode }, messageId)
+        .catch(error => {
+          showErrorAlert('Failed to move message', error.message);
+          props.navigation.goBack();
+        });
+    }
+    props.navigation.goBack();
+    Toast.show('Moved Message');
+  };
+
+  const handleNarrow = (pickedStreamId: number) => {
+    setStreamId(pickedStreamId);
+    const pickedStreamName = allStreams.find(s => s.stream_id === pickedStreamId)?.name;
+    setNarrow(streamNarrow(String(pickedStreamName)));
+  };
+
+  const handlePropagateMode = (propagatePickerIndex: number) => {
```

Comment:
So this function is to handle the propagate model while using the Picker component.

After I added the types for the propagate_mode in updateMessage API, the setPropagateMode function does not work with onValueChange function of the Picker component(line 202). 

Since there will be only three values, I used if-else statements based on the values of the index

Reply:
This works, but is fragile: these conditions in `handlePropagateMode` now have to line up with the order of options in the picker, and they're separated by quite a bit of code. It'd be easy for them to get out of sync, so that the items no longer had the effects they're labelled for.

For that reason it's better to use the `value` feature of Picker / PickerItem.

If you get errors from Flow but you're confident that what actually happens with the data is correct, you can add a `$FlowFixMe` comment with an explanation of why the actual behavior is correct.